### PR TITLE
ROC-3485: Remove names returned by postcode lookup from addresses

### PR DESCRIPTION
**Pull request by varma02**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-3485

### Change description
Format address based on residential or company address. 

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/main/public/js/postcode-lookup.js [hunk 1]
```diff
@@ -259,14 +259,27 @@
...
+        var postCode = lines[lines.length - 1].trim()
+
+        if (address.organisationName === "") {
+          valueFormattedAddress = [
+            lines[0].trim(),
+            lines.length > 3 ? lines[1].trim() : '',
+            lines.length > 4 ? lines[2].trim() : '',
+            townOrCity,
+            postCode
+          ].join(', ')
```
Comment (geek-soft) [inline src/main/public/js/postcode-lookup.js]: Please do `.join(', ')` before you set `option.value`.

**Comment by geek-soft** [comment]
It looks like if `organisationName` is present in postcode lookup result it is always appended to formatted address as a first line. Solution looks valid but requires extensive test.

Ordnance Survey API seems to return addresses in exactly the same way.

**Comment by varma02** [comment]
here are some postcodes used for testing - EX2 4LR, EX2 8DP, NN3 8JZ, EX4 3HP, EX1 1QL

File: src/integration-test/tests/citizen/claim/pages/organisation-details.ts [hunk 2]
```diff
@@ -40,12 +43,12 @@ export class OrganisationDetailsPage {
...
-    I.click(fields.address.enterManually)
-    I.fillField(fields.address.line1, address.line1)
-    I.fillField(fields.address.line2, address.line2)
-    I.fillField(fields.address.postcode, address.postcode)
-    I.fillField(fields.address.city, address.city)
+  enterAddressOnPostCodeLookUp (postCodeLookup): void {
+    I.fillField(fields.address.postcodeLookUp, postCodeLookup.postCode)
+    I.click(buttons.postCodeLookUp)
+    I.waitForVisible(fields.address.selectAddressList)
+    I.click(fields.address.selectAddressList)
```
[the remark this answers is not in the document]
Comment (varma02) [reply src/integration-test/tests/citizen/claim/pages/organisation-details.ts]: you are right, click not necessary. I have tested it.

File: src/integration-test/tests/citizen/claim/pages/company-details.ts [hunk 3]
```diff
@@ -40,11 +43,12 @@ export class CompanyDetailsPage {
...
-  enterAddress (address: Address): void {
-    I.fillField(fields.address.line1, address.line1)
-    I.fillField(fields.address.line2, address.line2)
-    I.fillField(fields.address.postcode, address.postcode)
-    I.fillField(fields.address.city, address.city)
+  enterAddressOnPostCodeLookUp (postCodeLookup): void {
+    I.fillField(fields.address.postcodeLookUp, postCodeLookup.postCode)
+    I.click(buttons.postCodeLookUp)
+    I.waitForVisible(fields.address.selectAddressList)
+    I.click(fields.address.selectAddressList)
```
[the remark this answers is not in the document]
Comment (varma02) [reply src/integration-test/tests/citizen/claim/pages/company-details.ts]: you are right, click not necessary. I have tested it.

File: src/integration-test/tests/citizen/claim/steps/claim.ts [hunk 4]
```diff
@@ -115,7 +115,7 @@ export class ClaimSteps {
       case PartyType.INDIVIDUAL:
         partyTypePage.selectIndividual()
         individualDetailsPage.enterName(defendant.name)
-        individualDetailsPage.enterAddressOnPostCodeLookUp(postCodeLookup)
+        individualDetailsPage.enterAddressOnPostCodeLookUp(postcodeLookupQuery)
```
Comment (geek-soft) [inline src/integration-test/tests/citizen/claim/steps/claim.ts]: Please rename to be `lookupAddress` as well to make it consistent.

Comment (varma02) [reply src/integration-test/tests/citizen/claim/steps/claim.ts]: done

**Pull request merged**